### PR TITLE
omcompiler 1.9.6 (new formula)

### PR DESCRIPTION
**Pull request by SimplyDanny**
### Have you:

- [x] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [x] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [x] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [x] Built your formula locally prior to submission with `brew install <formula>`?

### New formula: have you

- [x] Written a sensible test? (The best test is to compile and run a sample program.)

Modelica is an object-oriented, equation based language to conveniently model complex physical systems. OpenModelica is one implementation of the language providing a modeling and simulation environment. OMCompiler is the base of OpenModelica. It translates Modelica code into C code.

### CODE REVIEW

File: omcompiler.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
+class Omcompiler < Formula
+  desc "Modelica compiler translating Modelica to C code"
+  homepage "https://www.openmodelica.org"
+  url "https://github.com/OpenModelica/OMCompiler.git",
+      :tag => "v1.9.6",
+      :revision => "d3c35c90b382998275f42eb5da05c0f25e6b63a8"
+
+  # Options
+  option "with-boost",      "Use boost"
+  option "with-sundials",   "Use sundials to add more numerical solvers to the simulation runtime"
```
Comment (dpo) [inline omcompiler.rb]: These two options are redundant with lines 29 and 30, and should be removed.

File: omcompiler.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  depends_on "automake"     => :build
+  depends_on "cmake"        => :build
+  depends_on "libtool"      => :build
+  depends_on "pkg-config"   => :build
+  depends_on "gnu-sed"      => :build
+  depends_on "xz"           => :build
+
+  # Essential dependencies
+  depends_on :fortran
+  depends_on "gcc"
```
Comment (dpo) [inline omcompiler.rb]: There's no need to depend on `gcc`, the `:fortran` dependency takes care of it. If you mean that this library only builds with gcc and not with clang, then there should be a `fails_with` block.

File: omcompiler.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  depends_on "homebrew/science/lp_solve"
+  depends_on "homebrew/science/hwloc"
```
Comment (dpo) [inline omcompiler.rb]: There's no need to fully qualify formulae from this tap.

**Comment by SimplyDanny** [comment]
All annotations are put into practice. I hope I did not forget anything.

**Comment by dpo** [comment]
Thank you!!!

**Pull request closed without merging**